src/types/widget.ts

```diff
@@ -20,4 +20,4 @@ export interface Widget {
   getConfiguration?(previousConfiguration: SearchParameters): SearchParameters;
 }
 
-export type WidgetFactory<T> = (widgetParams: T) => Widget;
+export type WidgetFactory<WidgetParams> = (widgetParams: WidgetParams) => Widget;
```

Comment:
```suggestion
export type WidgetFactory<TWidgetParams> = (widgetParams: TWidgetParams) => Widget;
```

Keep the `T` to keep the information that it's a generic.